tests/fixtures.rs

```diff
@@ -11,10 +11,24 @@ fn fixtures() {
 
     for fixture in std::fs::read_dir(root_dir.join("tests/crates")).unwrap() {
         let fixture = fixture.unwrap();
+        if !fixture.file_type().unwrap().is_dir() {
+            continue;
+        }
         let fixture_path = fixture.path();
 
-        // FIXME: don't expect the crate to be in the `ui` subdir
-        let dir = fixture_path.join("ui");
+        let file = std::fs::File::open(fixture_path.with_extension("sub-path"))
+            .and_then(|mut file| {
+                use std::io::Read;
+                let mut contents = String::new();
+                file.read_to_string(&mut contents)?;
+                Ok(contents)
+            });
+        let dir = if let Ok(path) = file {
+            fixture_path.join(path.trim())
+        } else {
+            fixture_path.clone()
+        };
```

Comment:
So, we now have files that tell us which directory to look into for applying fixtures? Why not allow multiple sub paths to check? I'd probably start refactoring the whole nested for loop business with a bunch of flat_maps now going from crate folders to tests.